packages/sanity/src/core/preview/liveDocumentIdSet.ts

```diff
@@ -0,0 +1,112 @@
+import {type QueryParams, type SanityClient} from '@sanity/client'
+import {sortedIndex} from 'lodash'
+import {of} from 'rxjs'
+import {distinctUntilChanged, filter, map, mergeMap, scan, tap} from 'rxjs/operators'
+
+export type DocumentIdSetObserverState = {
+  status: 'reconnecting' | 'connected'
+  documentIds: string[]
+}
+
+interface LiveDocumentIdSetOptions {
+  insert?: 'sorted' | 'prepend' | 'append'
+}
+
+export function createDocumentIdSetObserver(client: SanityClient) {
+  return function observe(
+    queryFilter: string,
+    params?: QueryParams,
+    options: LiveDocumentIdSetOptions = {},
+  ) {
+    const {insert: insertOption = 'sorted'} = options
+
+    const query = `*[${queryFilter}]._id`
+    function fetchFilter() {
+      return client.observable
+        .fetch(`*[${queryFilter}]._id`, params, {
```

Comment:
COULD: use `query` here instead
```suggestion
        .fetch(query, params, {
```

Reply:
Oh, yes, that wasn't intentional. Thanks for catching!

Reply:
Fixed by 17ab291a61dc8c79e253ccf562d30b6f80be2a4e